### PR TITLE
moved cfg.merge to the cli interface

### PR DESCRIPTION
**Pull request by v1kko**
- [x] make merge a cli tool
- [x] write the documentation

### CODE REVIEW

**Comment by v1kko** [comment]
@stefsmeets just want to know if this merge interface is useful, or if you think that there are still options missing

File: src/duqtools/cli.py [hunk 1]
```diff
@@ -384,10 +384,54 @@ def cli_dash(**kwargs):
...
+@click.option('-t',
+              '--template',
+              required=True,
+              type=str,
+              help='IMAS location to use as the template for the target')
+@click.option('-T',
+              '--target',
+              required=True,
+              type=str,
+              help='IMAS location to store the result in')
```
Comment (stefsmeets) [inline src/duqtools/cli.py:396]: Would `-o` / `--out` be more logical here?

File: src/duqtools/cli.py [hunk 2]
```diff
@@ -384,10 +384,54 @@ def cli_dash(**kwargs):
...
+@click.option('-h',
+              '--handle',
+              'handles',
+              type=str,
+              help='handles to merge to the target',
+              multiple=True)
```
Comment (stefsmeets) [inline src/duqtools/cli.py:402]: In `duqtools plot` we use this as the interface:
```
  -m, --imas TEXT     IMAS path formatted as <user>/<db>/<shot>/<number>.
  -u, --user TEXT     IMAS user.
  -d, --db TEXT       IMAS database.
  -s, --shot INTEGER  IMAS shot.
  -r, --runs INTEGER  IMAS run (multiple runs can be specified)
  -i, --input TEXT    Input file, i.e. `data.csv` or `runs.yaml`
```

I think we should aim for consistency here. Maybe this tools should not have the option to define separate handles and just take a `data.csv` as input.

File: src/duqtools/merge.py [hunk 3]
```diff
@@ -11,34 +12,47 @@
...
 
-    handles = read_imas_handles_from_file(cfg.merge.data).values()
+    handles = [ImasHandle.from_string(handle) for handle in handles]
+    if runfile:
+        handles = handles + list(read_imas_handles_from_file(runfile).values())
+
+    handles = set(handles)  # Remove duplicate handles
 
     for handle in handles:
         op_queue.add(action=lambda: None,
```
Comment (stefsmeets) [inline src/duqtools/merge.py:28]: We can use `op_queue.add_no_op(...)` here. I changed a bunch of these in `main` just now, but maybe that still needs to be merged in.

File: src/duqtools/merge.py [hunk 4]
```diff
@@ -11,34 +12,47 @@
...
                      description='Merging all known variables')
     else:
-        variables = [var_lookup[name] for name in cfg.merge.variables]
-        for variable in variables:
-            op_queue.add(action=lambda: None,
-                         description='Merging variable',
-                         extra_description=f'{variable.name}')
+        if not variables:
+            op_queue.add_no_op('No variables specified for merge',
+                               'recommend to abort')
```
Comment (stefsmeets) [inline src/duqtools/merge.py:41]: Nothing to do, so just return and get out of there 😅

```suggestion
        if not variables:
            return
```

Comment (stefsmeets) [reply src/duqtools/merge.py:41]: This means you can get rid of the else block as well.

**Review by stefsmeets — APPROVED** [review]
Cool! Some minor changes, and I think we need to have a look at the API for the CLI tools at some point to make sure the interface is consistent.

File: src/duqtools/cli.py (hunk 2 above)
Comment (v1kko) [reply src/duqtools/cli.py:402]: I think `plot` should be updated then, it does not make sense to have seperate options for a single IMAS path

File: src/duqtools/merge.py (hunk 3 above)
Comment (v1kko) [reply src/duqtools/merge.py:28]: The default text color of `add_no_op` is red, and I wanted green, I can add the click format to green here ...

File: src/duqtools/cli.py (hunk 2 above)
Comment (v1kko) [reply src/duqtools/cli.py:402]: the effect of the '-r' flag can also be achieved with:
> -h jet/1234/80{01,02,03,04,05,06,07}

File: src/duqtools/merge.py (hunk 3 above)
Comment (stefsmeets) [reply src/duqtools/merge.py:28]: Alright, makes sense 👍

File: src/duqtools/cli.py (hunk 2 above)
Comment (stefsmeets) [reply src/duqtools/cli.py:402]: Sounds good, I will make an issue for this.

**Pull request merged**